### ISSUE TITLE
64-bit BAR Support

### ISSUE THREAD

**Issue by thomasbarrett** [nutanix/libvfio-user#710]
libvfio-user does not appear to support 64-bit BAR. In the 'private.h' file, I noticed that the region size was constrained to uint32_t. Even after fixing this, I can't seem to be able to create a 64-bit BAR. Does anyone have any experience using libvfio-user with 64-bit BARs?

**Comment by tmakatos** [comment]
We haven't come across 64-bit BARs so far. FWIU, it's should be just a matter of changing `uint32_t` to `uint64_t` in a few places. When you say it doesn't work, what do you mean? Do you have some logs/code to share?

**Comment by ramakristipati** [comment]
is this fixed?

**Comment by jlevon** [comment]
No, but we would definitely be interested in patches if you're interested in making this work

**Comment by jlevon** [comment]
@thomasbarrett @ramakristipati can you please try out https://github.com/nutanix/libvfio-user/pull/792 ?

**Comment by jlevon** [comment]
Basic 64-bit support done via https://github.com/nutanix/libvfio-user/pull/792
What remains is support for individual bars over 4g
